question.php

```diff
@@ -99,15 +100,34 @@ private function update_ui(attempt_ui $ui): void {
      *      being started. Can be used to store state.
      * @param int $variant which variant of this question to start. Will be between
      *      1 and {@see get_num_variants()} inclusive.
-     * @throws moodle_exception
+     * @throws Throwable
      */
     public function start_attempt(question_attempt_step $step, $variant): void {
-        $attempt = $this->api->package($this->packagehash, $this->packagefile)->start_attempt($this->questionstate, $variant);
+        global $PAGE;
+
+        try {
+            $attempt = $this->api->package($this->packagehash, $this->packagefile)->start_attempt($this->questionstate, $variant);
 
-        $this->attemptstate = $attempt->attemptstate;
-        $step->set_qt_var(constants::QT_VAR_ATTEMPT_STATE, $attempt->attemptstate);
-        $this->scoringstate = null;
-        $this->update_ui($attempt->ui);
+            $this->attemptstate = $attempt->attemptstate;
+            $step->set_qt_var(constants::QT_VAR_ATTEMPT_STATE, $attempt->attemptstate);
+            $this->scoringstate = null;
+            $this->update_ui($attempt->ui);
+        } catch (Throwable $t) {
+            // Trigger error event.
+            // TODO: get question_attempt-id via DB-query (question_attempt_step-id -> question_attempt-id)?
+            $params = [
+                'context' => $PAGE->context,
+                'relateduserid' => $step->get_user_id(),
+                'other' => [
+                    'questionid' => $this->id,
+                    'errormessage' => $t->getMessage(),
+                ],
+            ];
```

Comment:
Seit #135 könntest du via `$this->get_behaviour()->get_qa()->get_database_id()` da rankommen. In `start_attempt` ist der Attempt allerdings noch gar nicht persistiert, und hat entsprechend keine ID. Du könntest höchstens die Usage-ID und den Slot ausgeben, die sollten gemeinsam eindeutig einen Attempt identifizieren.